config/src/main/java/com/typesafe/config/Path.java

```diff
@@ -0,0 +1,21 @@
+package com.typesafe.config;
+
+/**
+ * The path of a reference. For a reference like ${foo.bar}, for instance, the
+ * path is ["foo", "bar"].
+ */
+public interface Path {
+
+    /**
+     * Returns the first element of this path, for instance for the path of
+     * ${foo.bar} the first element is "foo".
+     */
+    public String getFirst();
+
+    /**
+     * Returns the tail part of this path, for instance for the path of
+     * ${foo.bar.baz} the remainder is the path of ${bar.baz}.
+     */
+    public Path getRemainder();
+
+}
```

Comment:
I'd be inclined to let people use `ConfigUtils.splitPath` rather than making this public, since that's how the rest of the APIs have gone so far.